### PR TITLE
Dm 2060 keep build node images

### PR DESCRIPTION
**Pull request by amues**
## Checklist

- [X] Public interface changes have been documented in one of the example Jenkinsfiles.
- [X] Changes have been tested in the test branch with https://github.com/ess-dmsc/jenkins-shared-library-test

### Description of work

Add functionality to clean Docker images, keeping only images from the default build nodes. This should reduce the need to pull images from Docker Hub.

### Issue or JIRA ticket

Closes ECDC-2060.

### Acceptance criteria

I have tested the changes manually in the test branch.

### Unit tests

I have added some tests and tried to start making the library more testable.

### Other

n/a

---

## Code Review (To be filled in by the reviewer only)

- [ ] Is the code of an acceptable quality?
- [ ] Do the changes function as described and are they robust?
- [ ] Has the documentation been updated?
- [ ] Have associated JIRA tickets been updated?

### CODE REVIEW

File: src/ecdcpipeline/DockerWrapper.groovy [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+/**
+ * Wrapper for Docker commands.
+ */
+class DockerWrapper implements Serializable {
+
+  private def script
+  private def dockerOutputParser
+
+  /**
+   * <p></p>
```
Comment (mortenjc) [inline src/ecdcpipeline/DockerWrapper.groovy:14]: should there be something between the <>s? or could that line be deleted?

File: src/ecdcpipeline/ImageRemovalFilter.groovy [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ */
+class ImageRemovalFilter implements Serializable {
+
+  /**
+   * List of image names to be kept.
+   */
+  private def imageNamesToKeep
+
+  /**
+   * <p></p>
```
Comment (mortenjc) [inline src/ecdcpipeline/ImageRemovalFilter.groovy:14]: same comment

File: src/ecdcpipeline/ImageRemover.groovy [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+/**
+ * Container image removal.
+ */
+class ImageRemover implements Serializable {
+
+  private def dockerWrapper
+
+  /**
+   * <p></p>
```
Comment (mortenjc) [inline src/ecdcpipeline/ImageRemover.groovy:15]: ..

**Review by mortenjc — APPROVED** [review]

File: src/ecdcpipeline/DockerWrapper.groovy (hunk 1 above)
Comment (amues) [reply src/ecdcpipeline/DockerWrapper.groovy:14]: I used to believe this was necessary due to some issue that occurred if we didn't have it, but I don't remember what it was and could not see any difference in the documentation by removing it, so I removed it.

File: src/ecdcpipeline/ImageRemover.groovy (hunk 3 above)
Comment (amues) [reply src/ecdcpipeline/ImageRemover.groovy:15]: See above.

**Pull request merged**